### ISSUE TITLE
Grouped hints

### ISSUE THREAD

**Issue by hsoft** [fanglingsu/vimb#502]
When a page has a lot of labels, it can be cumbersome to reliably type a hint label, especially if it's a page that is often visited. It would be great if "chunks" of links could have a common, predictable prefix.

I began working on a POC where I group links based on whether they are children of a `<li>` element and I think it could work great. I'm thinking of prefixing all grouped hints with a letter (`group-hint-keys`?). For example, if a page has 3 `ul/li`, then hits would be `a1, a2, a3, ..., b1, b2, ..., c1,c2, ...`.

Later, we can think about making the grouping algorithm even smarter.

Before going further in my work, I was wondering what you thought of the idea @fanglingsu .

**Comment by fanglingsu** [comment]
It seems that I do not see such cases often, so the current behaviour works well for me. If you get this working and this does not need too much code - I'd merge it. It might be nice if this feature could be enabled/disabled at compile time.

**Comment by hsoft** [comment]
It seems to me that the cases are simply everywhere, but a good example I have are saved searches in bugzilla. They're at the bottom of the page. If I often use, for example, the 3rd saved search, I will have to look at the hint and type something like `66` or `85` for each page, and for each page the hint will be different (depending on the number of bugs currently displayed). With this feature, the number of `ul/li` in the page will normally stay the same all the time, so the group prefix would be stable, so I could use a more predictable `d3` (for example).

Anyways, I'll give it a try and we'll see how well it works.

**Comment by hsoft** [comment]
I made a very coarse attempt at implementing the feature at https://github.com/hsoft/vimb/tree/hints-groups but I'm a bit disappointed by the result. The implementation adds significant complexity but usage isn't greatly improved. Maybe it's not such a good idea after all.

**Comment by fanglingsu** [comment]
@hsoft I did not think that this coarse prototype could be done with such a little change. But I personally do not need this. If I use hinting, I'll type some of the text of the hinted elements to near down the result and than <tab><tab>.. or type the hintlabel.
What is the main issue you want to solve?
1. Are there overlapping hint labels,
2. or there to many hintlabels close together?

For 1. I have no goof idea at the time, for 2. it might be worth to rework you attempt to not change the label but the color of the labels based on the group. But I'm not sure if this would help.

**Comment by hsoft** [comment]
My use case was pages with a lot of links, as I wrote in my 2nd comment. Identifying labels like `68` and then typing them isn't super efficient. I wondered if grouping them logically could help. Again, as I wrote, results are inconclusive for now, I don't think I'll try this path further.

I haven't acquired the habit of selecting hints by text inside it. I'll try to train myself at that and see if I can get efficient.
